gomint-api/src/main/java/io/gomint/plugin/Plugin.java

```diff
@@ -115,28 +133,68 @@ public void onUninstall() {
      *
      * @return Plugin installation status
      */
-    public boolean isInstalled() {
-        return this.pluginManager.isPluginInstalled( this.name );
+    public final boolean isInstalled() {
+        return this.pluginManager.isPluginInstalled(this.name);
     }
 
     /**
      * Register a new command to this plugin
      *
      * @param command which should be registered
      */
-    public Plugin registerCommand( Command command ) {
-        this.pluginManager.registerCommand( this, command );
+    public final Plugin registerCommand(Command command) {
+        this.pluginManager.registerCommand(this, command);
+        return this;
+    }
+
+    /**
+     * Register a new event listener for this plugin.
+     *
+     * @param listener The listener which should be registered
+     * @see #registerListener(EventListener, Predicate)
+     * @see #registerActiveWorldsListener(EventListener)
+     * @since 2.0
+     */
+    public final Plugin registerListener(EventListener listener) {
+        this.pluginManager.registerListener(this, listener);
+        this.listeners.add(listener);
+        return this;
+    }
+
+    /**
+     * Register a new event listener for this plugin with the filter predicate
+     * {@linkplain #eventInActiveWorlds(Event)}.
+     * <br><br>
+     * Events implementing {@linkplain io.gomint.event.interfaces.WorldEvent WorldEvent} will be filtered for the given
+     * listener, it will only be called for events taking place in worlds which {@linkplain World#folder() folder name}
+     * is present in the {@code worlds.yml} configuration.
+     *
+     * @param listener The listener which should be registered
+     * @see #registerListener(EventListener)
+     * @see #registerListener(EventListener, Predicate)
+     * @since 2.0
+     */
+    public final Plugin registerActiveWorldsListener(EventListener listener) {
```

Comment:
Good idea

Reply:
I think I named this method so, because the registerListener method with the custom Predicate argument does add its own active worlds check. Should the method for listener registration without predicate be renamed to registerGlobalListener or removed?

Reply:
Can you use the same name but without the predicate argument?

Reply:
We currently have `registerListener(EventListener)`, `registerListener(EventListener, Predicate)` and `registerActiveWorldsListener(EventListener)`.
Can you explain what you want renamed exactly please, I'm a little confused.

Reply:
registerActiveWorldsListener(EventListener) -> registerListener(EventListener)

which then calls 

registerListener(EventListener, Predicate) with null which internally always has the world checking predicate and the one given from the plugin